src/components/shared/FilterNavigatorBar.js

```diff
@@ -9,23 +9,43 @@ import classNames from 'classnames';
 import { CSSTransition, TransitionGroup } from 'react-transition-group';
 import './FilterNavigatorBar.css';
 
+type FilterNavigatorBarButtonProps = {|
+  onClick: (number) => mixed,
+  index: number,
+  children: React.Node,
+|};
+
+class FilterNavigatorBarButton extends React.PureComponent<FilterNavigatorBarButtonProps> {
```

Comment:
I followed the pattern suggested in https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md#lists-of-items